code/go/0chain.net/miner/protocol_block_main.go

```diff
@@ -293,7 +293,8 @@ func (mc *Chain) generateBlock(ctx context.Context, b *block.Block,
 		}
 	}
 
-	if config.DevConfiguration.IsBlockRewards {
+	if config.DevConfiguration.IsBlockRewards &&
+		b.Round%config.SmartContractConfig.GetInt64("smart_contracts.storagesc.block_reward.challenge_period") == 0 {
```

Comment:
but @peterlimg we don't have access to storagesc.scConfig from here.

Reply:
Add method to the StateContextI to get the config, the config should be read from MPT, so it could be different from the value read from the viper. We should always use the loaded config instead of reading it from viper directly.